flink-core/src/main/java/org/apache/flink/configuration/SecurityOptions.java

```diff
@@ -33,34 +30,33 @@
 /**
  * The set of configuration options relating to security.
  */
-@PublicEvolving
-@ConfigGroups(groups = {
-	@ConfigGroup(name = "Kerberos", keyPrefix = "security.kerberos"),
```

Comment:
Why did you opt for replacing the ConfigGroups here, but kept them for `env` in `CoreOptions`?

Reply:
No specific reason. It is rework step by step. Not everything is migrated in the first step.
At the moment there are still parts that are generated by Class / ConfigGroup, while others are generated by Section option.

In the long run, I would suggest to do everything by section option, that is simpler. But I didn't want just do this without doing a this first rework and see how this model serves us.

Reply:
> section option [ are] simpler

This is debatable. The ConfigGroups were written on purpose the way they are so that devs can add new options without having to worry or even know how the docs are generated. You could just add a new kerberos option and it _would just work_.

Meanwhile, if a dev now adds an option they have to decide whether the option is common/expert, and if they forget to add any of the annotations are at risk of the option not being documented at all.

Reply:
The core issue I have here is that we introduce an inconsistency in how options are translated into include files, which make it impossible to implement any safeguards to prevent dev errors. In my example above, the generator won't complain about a new ssl option being added without a common/expert annotation, because it will still be added to the file we generate for each config class by default, and the generator assumes that all include files are used in the documentation.

If on the other hand we would completely remove config groups, and the default include file per config class, then we could mandate that every new option must have a section annotation. This we could easily enforce.